packages/@aws-cdk/aws-cognito/lib/user-pool.ts

```diff
@@ -0,0 +1,184 @@
+import lambda = require('@aws-cdk/aws-lambda');
+import cdk = require('@aws-cdk/cdk');
+import { CfnUserPool } from './cognito.generated';
+
+/**
+ * Standard attributes
+ */
+export enum Attribute {
+  ADDRESS = 'address',
+  BIRTHDATE = 'birthdate',
+  EMAIL = 'email',
+  FAMILY_NAME = 'family_name',
+  GENDER = 'gender',
+  GIVEN_NAME = 'given_name',
+  LOCALE = 'locale',
+  MIDDLE_NAME = 'middle_name',
+  NAME = 'name',
+  NICKNAME = 'nickname',
+  PHONE_NUMBER = 'phone_number',
+  PICTURE = 'picture',
+  PREFERRED_USERNAME = 'preferred_username',
+  PROFILE = 'profile',
+  TIMEZONE = 'timezone',
+  UPDATED_AT = 'updated_at',
+  WEBSITE = 'website'
+}
+
+export enum SignInType {
+  USERNAME,
+  EMAIL,
+  PHONE,
+  EMAIL_OR_PHONE
+}
+
+export interface UserPoolTriggers {
+  /**
+   * Creates an authentication challenge.
+   */
+  createAuthChallenge?: lambda.Function;
```

Comment:
Should want to document the expectations for the function, too. What is it's input, what should it output and how? A way to solve is to point to the relevant AWS Documentation page. That is true for the other triggers, too.